pkg/validator/controller.go

```diff
@@ -67,6 +69,37 @@ func ValidateControllers(config conf.Configuration, kubeResources *kube.Resource
 	}
 }
 
+func applyLimitRanges(controller *Controller, limits []corev1.LimitRange) {
+	for _, limit := range limits {
+		if limit.ObjectMeta.Namespace != controller.Namespace {
+			continue
+		}
+		for _, limitItem := range limit.Spec.Limits {
+			if limitItem.Type != corev1.LimitTypeContainer {
```

Comment:
Yeah I brought this up in my top-level comment. The `Pod` LimitRange only affects the amount of resources used by the pod as a whole, rather than an individual container. I think best practice would be to set limits at the container level, so we wouldn't want to let deployments get away with just a pod-level limit.

What do you think?

Reply:
Ah thank you, it helps to read the top level comment. I'll have to read some docs about the subtle differences between each

Reply:
Ok - i've played around with this locally and I'm getting some super weird behavior / logic that isn't abundantly clear. 

1. You can have multiple LimitRanges defined per namespace
2. For spec.Limits[].type == Container -- the Kubernetes behavior is to take the _lowest_ value of any of the limits set in the namespace. The current logic will just iterate all the limits and the last one in the list wins. 

I am confused by something here, when I use `kubectl get pod pod-name-1 -oyaml`, I am seeing the default values reflected in the resources of the container. It would appear this doesn't happen in the kubernetes List() we're doing?? Maybe we could isolate why the defaults are "realized" in the `kubectl` and not the API call?